pkg/controllers/types.go

```diff
@@ -31,6 +34,13 @@ type Controller interface {
 	// For returns a default instantiation of the resource and is injected by
 	// data from the API Server at the start of the reconciliation loop.
 	For() client.Object
+	// Watches returns the necessary information to create a watch
+	//   a. Source: the resource that is being watched
+	//   b. EventHandler: which controller objects to be reconciled
+	//   c. WatchesOption: which events can be filtered out before processed
+	Watches(context.Context) (source.Source, handler.EventHandler, builder.WatchesOption)
+	// ConcurrentReconciles controls the number of concurrent reconciles that can occur
```

Comment:
Could you add something around the Why here? I'm wondering what the intent of this is? (Maybe add a very simple representative example?)

Reply:
This is being removed